keras_nlp/utils/tensor_utils.py

```diff
@@ -46,20 +47,75 @@ def tensor_to_list(inputs):
         list_outputs = inputs.numpy()
         if inputs.shape.rank != 0:
             list_outputs = list_outputs.tolist()
+    if inputs.dtype == tf.string:
+        list_outputs = _decode_strings_to_utf8(list_outputs)
     return list_outputs
 
 
-def tensor_to_string_list(inputs):
-    """Detokenize and convert tensor to nested lists of python strings.
+def convert_to_backend_tensor_or_python_list(x):
+    """
+    Convert a tensor to the backend friendly representation of the data.
+
+    This wraps `ops.convert_to_tensor` to account for the fact that torch and
+    jax both lack native types for ragged and string data.
+
+    If we encounter one of these types in torch or jax, we will instead covert
+    the tensor to simple pythonic types (lists of strings).
+    """
+    if isinstance(x, tf.RaggedTensor) or x.dtype == tf.string:
+        return tensor_to_list(x)
+    return ops.convert_to_tensor(x)
+
 
-    This is a convenience method which converts each byte string to a python
-    string.
+def convert_to_ragged_batch(inputs):
+    """Convert pythonic or numpy-like input to a 2-D tf.RaggedTensor.
+
+    This is useful for text preprocessing layers which deal with already
+    tokenized or split text.
 
     Args:
-        inputs: Input tensor, or dict/list/tuple of input tensors.
+        inputs: A numpy-like input passed to a layer's call method. This input
```

Comment:
I'm confused why layers and layer-specific methods are mentioned here. Isn't this just a util?

Reply:
Good point. I will reword to be more general.